internal/runtime/runtime.go

```diff
@@ -355,6 +355,67 @@ func Source(unsupportedJSFeatures compat.JSFeature) logger.Source {
 			})
 		}
 
+		// These help for lowering async generator functions
+		export var __await = function (promise, isYieldStar) {
+			this[0] = promise
+			this[1] = isYieldStar
+		}
+		export var __asyncGenerator = (__this, __arguments, generator) => {
+			var resume = (k, v, yes, no) => {
+				try {
+					var x = generator[k](v), isAwait = (v = x.value) instanceof __await, done = x.done
+					Promise.resolve(isAwait ? v[0] : v)
+						.then(y => isAwait
+							? resume(k === 'return' ? k : 'next', v[1] ? { done: y.done, value: y.value } : y, yes, no)
+							: yes({ value: y, done }))
+						.catch(e => resume('throw', e, yes, no))
+				} catch (e) {
+					no(e)
+				}
+			}
+			var method = k => it[k] = x => new Promise((yes, no) => resume(k, x, yes, no))
+			var it = {}
+			return generator = generator.apply(__this, __arguments),
+				it[Symbol.asyncIterator] = () => it,
+				method('next'),
+				method('throw'),
+				method('return'),
+				it
+		}
+		export var __yieldStar = value => {
+			var obj = value[__knownSymbol('asyncIterator')]
```

Comment:
@evanw: Hi, I'm currently debugging a crash we're seeing after bumping esbuild to a more recent version where the only diff is the addition of this polyfill. Is this line meant to throw if `asyncIterator` is not defined/supported by the VM? What is then the reason for check on line 390 below?